api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -227,23 +235,16 @@ public void updateProfileForUser(DbUser user, Profile updatedProfile, Profile pr
     profileAuditor.fireUpdateAction(previousProfile, appliedUpdatedProfile);
   }
 
-  public void validateAndCleanProfile(Profile profile) throws BadRequestException {
-    // Validation steps, which yield a BadRequestException if errors are found.
-    String userName = profile.getUsername();
-    if (userName == null || userName.length() < 3 || userName.length() > 64) {
-      throw new BadRequestException(
-          "Username should be at least 3 characters and not more than 64 characters");
-    }
-    validateStringLength(profile.getGivenName(), "Given Name", 80, 1);
-    validateStringLength(profile.getFamilyName(), "Family Name", 80, 1);
-
+  public void cleanProfile(Profile profile) throws BadRequestException {
     // Cleaning steps, which provide non-null fields or apply some cleanup / transformation.
     profile.setDemographicSurvey(
         Optional.ofNullable(profile.getDemographicSurvey()).orElse(new DemographicSurvey()));
     profile.setInstitutionalAffiliations(
         Optional.ofNullable(profile.getInstitutionalAffiliations()).orElse(new ArrayList<>()));
-    // We always store the username as all lowercase.
-    profile.setUsername(profile.getUsername().toLowerCase());
+    if (profile.getUsername() != null) {
```

Comment:
should we not just bail here if the username is missing? Either throw IllegalStateException or return a failure code.